src/resolver.cpp

```diff
@@ -235,211 +222,95 @@ std::string UsdOpenAssetIOResolver::_CreateIdentifier(
-    logMsg << "[tid=" << std::this_thread::get_id() << "] "
-           << "\n  result.assetName: " << result.assetName
-           << "\n  result.repoPath: " << result.repoPath;
-    logger_->debug(logMsg.str());
-  }
-
-  return result;
+  return ArDefaultResolver::_GetAssetInfo(assetPath, resolvedPath);
 }
 
 ArTimestamp UsdOpenAssetIOResolver::_GetModificationTimestamp(
     const std::string &assetPath, const ArResolvedPath &resolvedPath) const {
-  const std::string fnName = TF_FUNC_NAME();
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] " << fnName
-           << "\n  assetPath: " << assetPath
-           << "\n  resolvedPath: " << resolvedPath.GetPathString();
-    logger_->debug(logMsg.str());
-  }
-
-  auto result = catchAndLogExceptions(
-      [&] {
-        if (manager_->isEntityReferenceString(assetPath)) {
-          // Deliberately use a valid fixed timestamp, which will force caching
-          // until we implement a more considered method.
-          // TODO(DF): Consider how best to handle this via OpenAssetIO.
-          return ArTimestamp{0};
-        }
-        return ArDefaultResolver::_GetModificationTimestamp(assetPath, resolvedPath);
-      },
-      logger_, fnName);
-
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] "
-           << "result: " << result.GetTime();
-    logger_->debug(logMsg.str());
-  }
-
-  return result;
+  return ArDefaultResolver::_GetModificationTimestamp(assetPath, resolvedPath);
 }
 
 std::shared_ptr<ArAsset> UsdOpenAssetIOResolver::_OpenAsset(
     const ArResolvedPath &resolvedPath) const {
-  const std::string fnName = TF_FUNC_NAME();
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] " << fnName
-           << "\n  resolvedPath: " << resolvedPath.GetPathString();
-    logger_->debug(logMsg.str());
-  }
-
-  auto result = catchAndLogExceptions(
-      [&] {
-        return ArDefaultResolver::_OpenAsset(
-            locationInManagerContextForEntity(manager_, readContext_, resolvedPath));
-      },
-      logger_, fnName);
-
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] "
-           << "result: " << result.get();
-    logger_->debug(logMsg.str());
-  }
-  return result;
+  return ArDefaultResolver::_OpenAsset(resolvedPath);
 }
 
-// TODO(DF): Implement for publishing workflow.
 bool UsdOpenAssetIOResolver::_CanWriteAssetToPath(const ArResolvedPath &resolvedPath,
                                                   std::string *whyNot) const {
-  auto result = ArDefaultResolver::CanWriteAssetToPath(resolvedPath, whyNot);
-
-  logger_->debug(TF_FUNC_NAME());
-  logger_->debug("  resolvedPath: " + resolvedPath.GetPathString());
-  logger_->debug("  result: " + std::to_string(static_cast<int>(result)));
-
-  return result;
+  return ArDefaultResolver::_CanWriteAssetToPath(resolvedPath, whyNot);
```

Comment:
[question] Just out of curiosity, do you know why this didn't cause us problems before now?

Reply:
We'd never tried opening a stage for write 🙃